frontend-react/src/TelemetryService.ts

```diff
@@ -6,6 +6,10 @@ import { getSessionMembershipState } from "./utils/SessionStorageTools";
 let reactPlugin: ReactPlugin | null = null;
 let appInsights: ApplicationInsights | null = null;
 
+export function getAppInsights() {
+    return appInsights;
```

Comment:
This should be worked into the `AppInsightsProvider`. Right now it's payload is `ai.reactPlugin` but if we provide the whole `ai` objects, we can destructure it in a component like `const { appInsights } = useAppInsightsContext()`

Reply:
My opinion is that it's fine to keep this as-is.  `appInsights` is something that's already initialized outside of React, and we don't need to tightly couple this instance with rendering.  Plus, this approach allows us to create functions that can be invoked inside or outside of React rendering whereas destructuring it from Context will prohibit that.

Reply:
Hm 🤔 good point, we wouldn't have to pipe those values from components into external functions. That considered, the antithetical question: do we even need the provider?? Might be something leftover from setup instructions that we don't utilize. My only concern is that I think using the hooks of the React AppInsights library might rely on the provider, but I haven't had time to look into that specific concern.

Reply:
My feeling about this sort of stuff is if we can extricate logic easily outside of rendering, we should do it.  There's no reason to tightly couple something that exists entirely outside the render cycle to React, and doing so would unnecessarily limit our ability to track analytics events outside of the Provider Context.

The way that the custom track event function is set up is simpler than using a hook too:
```jsx
// as-is, just call the track function
TrackAppInsightEvent(...);

// with hooks
const trackFn = useTrackEvent(...);
trackFn(...);
```

As it is right now, we also don't have to worry about the track function reference changing between renders since it's defined once for the entirety of the application lifespan.  The reference for the return function from `useTrackEvent` could potentially change, though to be frank I haven't personally confirmed that.

Ultimately, I don't think it hurts to keep the AppInsightsProvider in the render tree, but it doesn't really _provide_ (:trollface:) much value in my opinion.